packages/flutter/test/material/expansion_tile_initial_state_test.dart

```diff
@@ -72,5 +88,17 @@ void main() {
     expect(getHeight(topKey), getHeight(expandedKey) - 2.0);
     expect(getHeight(topKey), getHeight(collapsedKey) - getHeight(tileKey) - 2.0);
     expect(getHeight(topKey), getHeight(defaultKey) - getHeight(tileKey) - 2.0);
+
+    // Expanded should be collapsed now.
+    expandedContainerDecoration = getContainer(expandedKey).decoration;
+    expect(expandedContainerDecoration.color, Colors.transparent);
+    expect(expandedContainerDecoration.border.top.color, Colors.transparent);
+    expect(expandedContainerDecoration.border.bottom.color, Colors.transparent);
+
+    // Collapsed should be expanded now.
+    collapsedContainerDecoration = getContainer(collapsedKey).decoration;
+    expect(collapsedContainerDecoration.color, Colors.transparent);
+    expect(collapsedContainerDecoration.border.top.color, dividerColor);
+    expect(collapsedContainerDecoration.border.bottom.color, dividerColor);
   });
 }
```

Comment:
this test doesn't seem to check that the animation animates with the right color (i.e. that the original bug is actually fixed); am I missing something?

Reply:
You are right. This only checks in the initial and final state to make sure my change did not introduce a regression.

I need your guidance on how to add a mid point check.

Would `pump(1/2 animation duration)` produce a stable state that can be tested?

Reply:
Yup, pumps with fixed durations will very reliably get you to the exact position in time that you want.